packages/simplr-forms-dom/src/components/select.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { FieldValue } from "simplr-forms/contracts";
+import { DomFieldProps } from "../contracts/field";
+
+import { BaseDomField, BaseDomFieldState } from "../abstractions/base-dom-field";
+import { FieldOnChangeCallback } from "../contracts/field";
+import {
+    HTMLElementProps
+} from "../contracts/field";
+import {
+    FormProps
+} from "../contracts/form";
+
+export type SelectValue = string | string[];
+export type SelectOnChangeCallback = FieldOnChangeCallback<HTMLSelectElement>;
+
+/**
+ * Override the differences between extended interfaces.
+ */
+export interface SelectProps extends DomFieldProps, HTMLElementProps<HTMLSelectElement> {
+    name: string;
+    onFocus?: React.FocusEventHandler<HTMLSelectElement>;
+    onBlur?: React.FocusEventHandler<HTMLSelectElement>;
+    onChange?: SelectOnChangeCallback;
+
+    defaultValue?: SelectValue;
+    initialValue?: SelectValue;
+    value?: SelectValue;
+    ref?: React.Ref<Select>;
+}
+
+export interface SelectState extends BaseDomFieldState {
+    RenderValue: SelectValue;
+}
+
+export class Select extends BaseDomField<SelectProps, SelectState> {
+    protected get RawInitialValue(): SelectValue | undefined {
+        if (this.props.multiple ||
+            this.props.initialValue != null) {
+            return this.props.initialValue;
+        }
+        // If select is not multiple, we need to get first option value.

```

Comment:
If select does not have multiple options, then we need to get the first option value.